tests/func/test_fs.py

```diff
@@ -507,3 +509,59 @@ def test_download_dir_callback(tmp_dir, dvc, cloud):
     assert callback.size == 2
     assert callback.value == 2
     assert (tmp_dir / "dir").read_text() == {"foo": "foo", "bar": "bar"}
+
+
+@pytest.mark.parametrize("fs_type", ["git", "dvc"])
+def test_download_callbacks_on_dvc_git_fs(tmp_dir, dvc, scm, fs_type):
+    gen = tmp_dir.scm_gen if fs_type == "git" else tmp_dir.dvc_gen
+    gen({"dir": {"foo": "foo", "bar": "bar"}, "file": "file"}, commit="gen")
+
+    fs = dvc.dvcfs if fs_type == "dvc" else scm.get_fs("HEAD")
+
+    callback = fsspec.Callback()
+    fs.download(tmp_dir / "file", tmp_dir / "file2", callback=callback)
+
+    size = os.path.getsize(tmp_dir / "file")
+    assert (tmp_dir / "file2").read_text() == "file"
+    assert callback.size == size
+    assert callback.value == size
+
+    if fs_type == "git":
+        pytest.skip("gitfs does not support download_dir")
```

Comment:
`download_dir` is not working, not sure if we use `fs.download_dir` for git filesystems.